### PR TITLE
OPS-2635: setting up storage sets, migrations for snuba-access-logs

### PR DESCRIPTION
**Pull request by mwarkentin**
### OPS-2635

~~Depends on #3451~~

### CODE REVIEW

**Comment by evanh** [comment]
I ran this locally (and fixed a few some errors). The migration runs correctly as far as I can tell.

I think this is ready for regular review.

**Comment by mwarkentin** [comment]
@evanh looks like I might need to define some clusters settings?

https://github.com/getsentry/snuba/actions/runs/3549121456/jobs/5961125396

**Comment by evanh** [comment]
New bug I'm trying to figure out.

`DB::Exception: Engine Distributed doesn't support TTL clause. Currently only the following engines have support for the feature: [MergeTree, ReplicatedVersionedCollapsingMergeTree, ReplacingMergeTree, ReplicatedSummingMergeTree, ReplicatedAggregatingMergeTree, ReplicatedCollapsingMergeTree, ReplicatedGraphiteMergeTree, ReplicatedMergeTree, ReplicatedReplacingMergeTree, VersionedCollapsingMergeTree, SummingMergeTree, GraphiteMergeTree, CollapsingMergeTree, AggregatingMergeTree]. Stack trace:
`

**Comment by mwarkentin** [comment]
I think that makes sense, the distributed table wouldn't have data to expire. Guess we need to modify or duplicate the column definition to handle both engine types?

File: snuba/settings/__init__.py [hunk 1]
```diff
@@ -227,9 +228,13 @@ class RedisClusters(TypedDict):
 COLUMN_SPLIT_MAX_LIMIT = 1000
 COLUMN_SPLIT_MAX_RESULTS = 5000
 
-# The migration groups that can be skipped are listed in OPTIONAL_GROUPS.
-# Migrations for skipped groups will not be run.
-SKIPPED_MIGRATION_GROUPS: Set[str] = {"querylog", "profiles", "functions"}
+# Migrations in skipped groups will not be run
+SKIPPED_MIGRATION_GROUPS: Set[str] = {
```
Comment (lynnagara) [inline snuba/settings/__init__.py:232]: I think access_logs needs to be on the OPTIONAL_GROUPS list for skipping to be enabled on that group

**Comment by evanh** [comment]
`DB::Exception: Sharding expression has type FixedString(32), but should be one of integer type. Stack trace:`

@mwarkentin I think you need a different sharding key.

**Pull request closed without merging**